update tests generation code

benoit-pierre opened the pull request:
- fix C warnings (missing "string.h" include)
- use random instead of deprecated whrandom (and use an hard-coded seed so generated output is stable)
- use assertEqual instead of deprecated assert_
- improve assertions messages for binary checks
- automatically skip test with wrong endianness
- fix invalid use of '//' in generated C code

Note: the last commit update `test_events_le.py` and `test_requests_le.py` with the resulting generated output, someone with access to a big-endian system is needed to generate and update the big-endian variants.

benoit-pierre commented:
On the subject of tests, maybe they should be marked with `unittest.expectedFailure` when running under Python 3 for now, so the continuous integration builds pass?

Also, is there an official discussion channel (like an IRC channel)?

benoit-pierre commented:
So apparently Python 2.6 does not support `unittest.SkipTest`... Is support for such an old version really needed?

Also, wouldn't it be better to run some of the tests (`examples/...`, `test/test_xlib_display`) with a fake server? I expect to be able to run the same command as for Travis CI to check my code, but `test_xlib_display.test_set_get_pointer_mapping` is not nice and mess my mouse buttons...

vasily-v-ryabov commented:
Hi @benoit-pierre, sorry for delayed response. Some re-design of the library is really necessary, any help in this direction is great. I'll review all the changes during Monday. I also appreciate your efforts in Py3 compatibility fixes.

There is a mailing list: python-xlib-users@lists.sourceforge.net ([archive is here](https://sourceforge.net/p/python-xlib/mailman/python-xlib-users/))
I think IRC channel did exist some time ago. Did try to check it out yet. The most recent info can be found in PRs and issues in the GitHub repo. Peter is also watching the repo and sometimes he can reply.

benoit-pierre commented:
@vasily-v-ryabov: great, thanks!

I was about to fork the project when I discovered this repository. Personally, most of what I'm interested is related to improving [Plover](https://github.com/openstenoproject/plover). Here is what I have working / am working on (higher priority first):
- better XInput support: manually adding support for those missing requests needed for Plover, see [my branch](https://github.com/python-xlib/python-xlib/compare/master...benoit-pierre:bpierre)
- Python 3 support
- support for more extensions, particularly XKB: I'm working on a version that uses the XCB protocol definition XML files (via the `xcbgen` Python module provided by `xcb-proto`) to generate support for any supported extension

vasily-v-ryabov commented on test/__init__.py:
```diff
@@ -0,0 +1,79 @@
+    def assertBinaryEqual(self, bin1, bin2):
+        if bin1 == bin2:
+            return
+        self.fail('binary contents differ:\n' + bindiff(bin1, bin2))
```
Why not just `if bin1 != bin2: self.fail(...)`? It avoids explicit empty return and code is more compact. I would also prefer having PEP8 compatible names where possible (`assert_binary_equal` instead of `assertBinaryEqual` for method names). All these things will add more code quality warnings (we have QuantifiedCode and Landscape.io cloud checks for the repo). Just click `Details` link (in `Checks` section of a PR) to see new code quality warnings (QC may incorrectly mark old warnings as new ones in some cases - I had filed a bug report about it).

benoit-pierre replied:
> Why not just if bin1 != bin2: self.fail(...)? It avoids explicit empty return and code is more compact.

OK, amended.

> I would also prefer having PEP8 compatible names where possible (assert_binary_equal instead of assertBinaryEqual for method names).

This was in keeping with the style used by unittest for assert methods, something PEP8 mention is preferred (consistency over strict adherence).

vasily-v-ryabov replied:
> This was in keeping with the style used by unittest for assert methods

Yeah, I've already figured it out. Let's keep it consistent. You're right.

vasily-v-ryabov commented:
Older version of python-xlib can be used with Py2.6 if any. So I think we could drop this Python version support for the new release.

I don't like an idea to maintain expected failures list. Py3 red tasks is a reminder and additional motivation to fix them. :)

vasily-v-ryabov commented:
The rest of changes look fine to me. :+1:

benoit-pierre commented:
> Older version of python-xlib can be used with Py2.6 if any. So I think we could drop this Python version support for the new release.

OK, I dropped Python 2.6 support from Travis build matrix.

I also added a fix for `test_xlib_display` messing with the pointer mapping.

Let me know if the new changes are okay, and I'll rewrite the history before you merge.
